massgen/v2/team_factories.py

```diff
@@ -0,0 +1,672 @@
+
+Your expertise includes:
+- Market research and competitive analysis
+- Target audience identification and segmentation
+- Marketing strategy and campaign planning
+- Performance metrics and KPI definition
+- Channel strategy and media planning
+- Budget allocation and ROI optimization
+
+Focus on data-driven strategic insights that maximize marketing effectiveness and business impact.""",
+        tags=["strategy", "marketing", "research", "planning"]
+    )
+    
+    # Create agents
+    for config in [creative_director_config, copywriter_config, designer_config, strategist_config]:
+        agents[config.agent_id] = create_simple_agent(
+            agent_id=config.agent_id,
+            model=config.model,
+            api_key=config.api_key,
+            system_message=config.system_message,
+            session_id=config.session_id
+        )
+    
+    # Create orchestrator
+    orchestrator_config = OrchestratorConfig(
+        orchestrator_id=team_id,
+        max_duration=1200,  # 20 minutes for creative tasks
+        max_rounds=8,
+        voting_config={
+            "include_vote_counts": False,
+            "include_vote_reasons": True,
+            "anonymous_voting": False,
+            "voting_strategy": "simple_majority",
+            "tie_breaking": "random"  # Creative decisions can be subjective
+        }
+    )
+    
+    return create_orchestrator(
+        agents=agents,
+        orchestrator_id=team_id,
+        config=orchestrator_config.to_dict(),
+        session_id=session_id,
+        **kwargs
+    )
+
+
+def create_custom_team(
+    team_config: Dict[str, Any],
+    session_id: Optional[str] = None,
+    **kwargs
+) -> Orchestrator:
+    """
+    Create a custom team from configuration.
+    
+    Args:
+        team_config: Team configuration dictionary
+        session_id: Optional session ID for tracking
+        **kwargs: Additional configuration for orchestrator
+        
+    Returns:
+        Orchestrator: Configured custom team
+        
+    Example team_config:
+    {
+        "team_id": "custom_team",
+        "agents": [
+            {
+                "agent_id": "agent1",
+                "model": "gpt-4o",
+                "system_message": "You are...",
+                "tags": ["tag1", "tag2"]
+            }
+        ],
+        "orchestrator": {
+            "max_duration": 600,
+            "voting_config": {...}
+        }
+    }
+    """
+    team_id = team_config.get("team_id", "custom_team")
+    agent_configs = team_config.get("agents", [])
+    orchestrator_config = team_config.get("orchestrator", {})
+    
+    if not agent_configs:
+        raise ValueError("Team configuration must include at least one agent")
+    
+    # Create agents
+    agents = {}
+    for agent_data in agent_configs:
+        config = AgentConfig.from_dict({**agent_data, "session_id": session_id})
+        agents[config.agent_id] = create_simple_agent(
+            agent_id=config.agent_id,
```

Comment:
Just one minor thing: the current call to create_simple_agent only passes a fixed set of arguments (e.g., agent_id, model, system_message, etc.).

To support additional `SimpleAgent` parameters like `max_retries` (which is also specified in `development_team.yaml`, it might be helpful to pass through the remaining fields in `agent_data` via `**kwargs`.

Other than that, everything looks good to me!

Reply:
Thanks. Revised. PTAL.